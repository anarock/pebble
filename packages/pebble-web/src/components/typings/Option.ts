import { ControlProps } from "./Control";
import { Omit } from "utility-types";
import * as React from "react";
import { Interpolation } from "./emotionCustom";

export interface OptionProps<OptionType>
  extends Omit<ControlProps<OptionType>, "checked" | "type"> {
  isActive?: boolean;
  isSelected?: boolean;
  multiSelect?: boolean;
  labelStyles?: Interpolation;
  rightElement?: (props: OptionProps<OptionType>) => React.ReactNode;
}
