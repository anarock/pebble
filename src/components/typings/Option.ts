import { ControlProps } from "./Control";
import { Omit } from "utility-types";
import * as React from "react";

export interface OptionProps<OptionType>
  extends Omit<ControlProps<OptionType>, "checked" | "type"> {
  isActive?: boolean;
  isSelected?: boolean;
  multiSelect?: boolean;
  labelClassName?: string;
  rightElement?: (props: OptionProps<OptionType>) => React.ReactNode;
}
