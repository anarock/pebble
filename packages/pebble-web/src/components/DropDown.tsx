import * as React from "react";
import { DropdownProps, DropdownState } from "./typings/Dropdown";
import { DropDownButton } from "./Button";
import { dropDownStyle, wrapperStyle } from "./styles/Dropdown.styles";

import OutsideClick from "./OutsideClick";
import { animated } from "react-spring/renderprops.cjs";
import MountTransition from "./shared/MountTransition";
import { Manager, Reference, Popper } from "react-popper";
import { colors } from "pebble-shared";

class DropDown extends React.PureComponent<DropdownProps, DropdownState> {
  state: DropdownState = {
    isOpen: !!this.props.initiallyOpen
  };

  static defaultProps: Partial<DropdownProps> = {
    closeOnOutsideClick: true,
    placement: "bottom-start",
    modifiers: {
      hide: {
        enabled: false
      },
      preventOverflow: {
        enabled: false
      },
      flip: {
        enabled: false
      }
    }
  };

  private toggleDropdown = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  render() {
    const {
      buttonLabel,
      children,
      labelComponent,
      padding,
      styles,
      dropDownStyles,
      isSelected,
      disabled,
      labelStyles,
      onOutsideClick
    } = this.props;
    const { isOpen } = this.state;

    return (
      <OutsideClick
        styles={[wrapperStyle, styles]}
        onOutsideClick={() => {
          this.setState({
            isOpen: false
          });
          if (onOutsideClick) onOutsideClick(isOpen);
        }}
        disabled={!isOpen}
      >
        <Manager>
          <Reference>
            {({ ref }) => (
              <div style={{ display: "inline-block", width: "100%" }} ref={ref}>
                {labelComponent ? (
                  labelComponent({
                    isOpen,
                    toggleDropdown: this.toggleDropdown
                  })
                ) : (
                  <DropDownButton
                    isSelected={!!isSelected}
                    isOpen={isOpen}
                    onClick={this.toggleDropdown}
                    disabled={disabled}
                    styles={labelStyles}
                  >
                    {buttonLabel}
                  </DropDownButton>
                )}
              </div>
            )}
          </Reference>

          {/* TODO: Add native flag. */}
          <MountTransition visible={isOpen}>
            {transitionStyles => (
              <animated.div
                css={[dropDownStyle, dropDownStyles]}
                style={{ padding, ...transitionStyles }}
              >
                <Popper {...this.props} positionFixed>
                  {({ ref, style, placement, arrowProps }) => {
                    const popperWrapperStyle = {
                      ...style,
                      ...transitionStyles,
                      backgroundColor: colors.white.base,
                      transform: `${style.transform ||
                        ""} ${transitionStyles.transform || ""}`,
                      transformOrigin: `${arrowProps.style.left ||
                        0}px ${arrowProps.style.top || 0}px`,
                      padding
                    };

                    return (
                      <div
                        css={[dropDownStyle, dropDownStyles]}
                        ref={ref}
                        style={popperWrapperStyle}
                        data-placement={placement}
                      >
                        {children({
                          toggle: this.toggleDropdown,
                          isOpen: this.state.isOpen
                        })}
                      </div>
                    );
                  }}
                </Popper>
              </animated.div>
            )}
          </MountTransition>
        </Manager>
      </OutsideClick>
    );
  }
}

export default DropDown;
