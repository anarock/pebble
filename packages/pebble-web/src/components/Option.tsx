import * as React from "react";
import { OptionProps } from "./typings/Option";
import Control from "./shared/Control";
import {
  activeRow,
  rowWrapper,
  selectedRow,
  labelWrap
} from "./styles/Options.styles";
import Ink from "react-ink";
import { colors } from "pebble-shared";

const defaultProps = {
  rightElement: ({
    isSelected,
    multiSelect
  }: {
    isSelected: boolean;
    multiSelect: boolean;
  }) => {
    const iconClass = [
      "pi",
      isSelected && "pi-checkbox-selected",
      !isSelected && "pi-checkbox-unselected"
    ]
      .filter(Boolean)
      .join(" ");
    return multiSelect ? (
      <i
        className={iconClass}
        css={{
          marginLeft: "10px",
          color: isSelected ? colors.violet.base : colors.gray.light,
          fontSize: "20px"
        }}
      />
    ) : null;
  }
};

class Option<OptionType> extends React.Component<
  OptionProps<OptionType> & Required<typeof defaultProps>
> {
  static defaultProps = defaultProps;
  render() {
    const {
      label,
      isActive,
      isSelected,
      rightElement,
      labelStyles,
      styles
    } = this.props;
    const _class = [
      rowWrapper,
      !!isActive && activeRow,
      !!isSelected && selectedRow,
      styles
    ];
    return (
      <Control
        {...this.props}
        checked={this.props.isSelected}
        type={this.props.multiSelect ? "checkbox" : "radio"}
        styles={_class}
      >
        {() => {
          return (
            <>
              <div css={[labelWrap, labelStyles]}>{label}</div>
              {rightElement(this.props)}
              <Ink />
            </>
          );
        }}
      </Control>
    );
  }
}

export default Option;
