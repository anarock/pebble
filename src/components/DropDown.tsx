import * as React from "react";
import { DropdownProps, DropdownState } from "./typings/Dropdown";
import { DropDownButton } from "./Button";
import { dropDownStyle, wrapperStyle } from "./styles/Dropdown.styles";
import { cx } from "emotion";
import OutsideClick from "./OutsideClick";
import { animated } from "react-spring";
import MountTransition from "./shared/MountTransition";

class DropDown extends React.PureComponent<DropdownProps, DropdownState> {
  state: DropdownState = {
    isOpen: !!this.props.initiallyOpen
  };

  static defaultProps: Partial<DropdownProps> = {
    closeOnOutsideClick: true
  };

  private toggleDropdown = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  render() {
    const {
      buttonLabel,
      children,
      labelComponent,
      padding,
      className,
      dropDownClassName,
      isSelected,
      disabled,
      labelClassName,
      onOutsideClick
    } = this.props;
    const { isOpen } = this.state;

    return (
      <OutsideClick
        className={cx(wrapperStyle, className)}
        onOutsideClick={() => {
          this.setState({
            isOpen: false
          });
          if (onOutsideClick) onOutsideClick(isOpen);
        }}
        disabled={!isOpen}
      >
        {labelComponent ? (
          labelComponent({ isOpen, toggleDropdown: this.toggleDropdown })
        ) : (
          <DropDownButton
            isSelected={!!isSelected}
            isOpen={isOpen}
            onClick={this.toggleDropdown}
            disabled={disabled}
            className={labelClassName}
          >
            {buttonLabel}
          </DropDownButton>
        )}
        <MountTransition visible={isOpen} native>
          {transitionStyles => (
            <animated.div
              className={cx(dropDownStyle, dropDownClassName)}
              style={{ padding, ...transitionStyles }}
            >
              {children({ toggle: this.toggleDropdown, isOpen })}
            </animated.div>
          )}
        </MountTransition>
      </OutsideClick>
    );
  }
}

export default DropDown;
