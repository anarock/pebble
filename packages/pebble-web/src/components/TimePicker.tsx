import * as React from "react";
import DropDown from "./DropDown";
import Option from "./Option";
import OptionGroupRadio from "./OptionGroupRadio";
import css from "@emotion/css";
import {
  timePickerWrap,
  timePickerSelected,
  hourPicker,
  minutePicker,
  optionStyle,
  seperator
} from "./styles/TimePicker.styles";
import { TimePickerProps } from "./typings/TimePicker";
import { colors } from "pebble-shared";

const HOURS = /*#__PURE__*/ [...Array(12)].map((_, i) =>
  ("00" + (i + 1).toString(10)).slice(-2)
);
const MINUTES = /*#__PURE__*/ [...Array(4)].map((_, i) =>
  ("00" + (i * 15).toString(10)).slice(-2)
);

export const iconStyle = css({
  marginLeft: 15,
  fontWeight: "bold",
  fontSize: 8,
  transition: "transform ease-out .2s",
  willTransform: "transform",
  marginTop: 2,
  color: colors.gray.dark,
  display: "inline-block"
});

export const buttonStyle = css({
  padding: "15px 20px",
  fontSize: "12px",
  color: colors.gray.darker,
  lineHeight: "9px"
});

const modifiers = {
  preventOverflow: {
    enabled: false
  },
  flip: {
    enabled: false
  }
};

const TimePicker: React.FunctionComponent<TimePickerProps> = props => {
  const { selectedHour, selectedMinute, onHourChange, onMinuteChange } = props;
  const selected = !!selectedHour || selectedMinute !== undefined;

  return (
    <div css={[timePickerWrap, selected && timePickerSelected]}>
      <DropDown
        labelStyles={hourPicker}
        isSelected={selected}
        labelComponent={({ isOpen, toggleDropdown }) => (
          <div
            onClick={() => {
              toggleDropdown();
            }}
            css={buttonStyle}
            data-test-id="hour-label"
          >
            <span css={css({ marginRight: "15px" })}>
              {selectedHour ? selectedHour : "Hrs"}
            </span>
            <span>
              <i
                className="pi pi-arrow-drop-down"
                css={iconStyle}
                style={{ transform: isOpen ? "rotate(180deg)" : "none" }}
              />
            </span>
          </div>
        )}
        placement="bottom-start"
        modifiers={modifiers}
      >
        {({ toggle }) => (
          <OptionGroupRadio
            onChange={value => {
              // `value as number` is a escape hack because we have to handle number | string
              // and OptionGroupRadio cannot determine its type on its own.
              // TODO: Remove all `as`.
              onHourChange(value as number);
              toggle();
            }}
            selected={selectedHour}
            styles={optionStyle}
          >
            {HOURS.map(hour => (
              <Option
                key={hour}
                value={parseInt(hour, 10)}
                label={hour}
                styles={optionStyle}
              />
            ))}
          </OptionGroupRadio>
        )}
      </DropDown>
      <span css={seperator}>:</span>
      <DropDown
        labelStyles={minutePicker}
        isSelected={selected}
        labelComponent={({ isOpen, toggleDropdown }) => (
          <div
            onClick={() => {
              toggleDropdown();
            }}
            css={buttonStyle}
            data-test-id="minute-label"
          >
            <span css={css({ marginRight: "15px" })}>
              {selectedMinute !== undefined ? selectedMinute : "mins"}
            </span>
            <i
              className="pi pi-arrow-drop-down"
              css={iconStyle}
              style={{ transform: isOpen ? "rotate(180deg)" : "none" }}
            />
          </div>
        )}
        placement="bottom-start"
        modifiers={modifiers}
      >
        {({ toggle }) => (
          <OptionGroupRadio
            onChange={value => {
              // `value as number` is a escape hack because we have to handle number | string
              // and OptionGroupRadio cannot determine its type on its own.
              // TODO: Remove all `as`.
              onMinuteChange(value as number);
              toggle();
            }}
            selected={selectedMinute}
            styles={optionStyle}
          >
            {MINUTES.map(min => (
              <Option
                key={min}
                value={parseInt(min, 10)}
                label={min}
                styles={optionStyle}
              />
            ))}
          </OptionGroupRadio>
        )}
      </DropDown>
    </div>
  );
};

export default TimePicker;
