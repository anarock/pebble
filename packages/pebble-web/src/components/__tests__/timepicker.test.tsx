import renderer from "react-test-renderer";
import combos from "combos";
import TimePicker from "../TimePicker";
import * as React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import Option from "../Option";
import DropDown from "../DropDown";

function noop() {}

describe("TimePicker", () => {
  const _props = {
    selectedHour: [1, 12, undefined],
    selectedMinute: [0, 15, undefined]
  };

  const _combos = combos(_props);

  test.each(_combos)("%o", props => {
    const timePicker = renderer.create(
      <TimePicker {...props} onHourChange={noop} onMinuteChange={noop} />
    );

    const tree = timePicker.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("TimePicker: functionality test", () => {
  test("should call onHourChange/onMinuteChange on click", () => {
    const spyHour = sinon.spy();
    const spyMinute = sinon.spy();
    const clock = sinon.useFakeTimers();

    const timePicker = mount(
      <TimePicker onHourChange={spyHour} onMinuteChange={spyMinute} />
    );
    timePicker
      .find("[data-test-id='hour-label']")
      .hostNodes()
      .simulate("click");

    clock.tick(10000);

    timePicker
      .find(DropDown)
      .at(0)
      .find(Option)
      .at(0)
      .simulate("click");

    expect(spyHour.calledOnce).toBeTruthy();
    expect(spyHour.calledWith(1)).toBeTruthy();

    timePicker
      .find("[data-test-id='minute-label']")
      .hostNodes()
      .simulate("click");

    clock.tick(10000);

    timePicker
      .find(DropDown)
      .at(1)
      .find(Option)
      .at(0)
      .simulate("click");

    expect(spyMinute.calledOnce).toBeTruthy();
    expect(spyMinute.calledWith(0)).toBeTruthy();
  });
});
