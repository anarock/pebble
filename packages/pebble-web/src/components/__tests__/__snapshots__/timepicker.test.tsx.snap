// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker { selectedHour: 1, selectedMinute: 0 } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          1
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          0
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: 1, selectedMinute: 15 } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          1
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          15
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: 1, selectedMinute: undefined } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          1
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          mins
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: 12, selectedMinute: 0 } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          12
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          0
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: 12, selectedMinute: 15 } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          12
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          15
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: 12, selectedMinute: undefined } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          12
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          mins
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: undefined, selectedMinute: 0 } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          Hrs
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          0
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: undefined, selectedMinute: 15 } 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EDEDED;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          Hrs
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          15
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker { selectedHour: undefined, selectedMinute: undefined } 1`] = `
.emotion-3 {
  position: relative;
}

.emotion-2 {
  padding: 15px 20px;
  font-size: 12px;
  color: #101721;
  line-height: 9px;
}

.emotion-0 {
  margin-right: 15px;
}

.emotion-1 {
  margin-left: 15px;
  font-weight: bold;
  font-size: 8px;
  -webkit-transition: -webkit-transform ease-out .2s;
  -webkit-transition: transform ease-out .2s;
  transition: transform ease-out .2s;
  will-transform: transform;
  margin-top: 2px;
  color: #6B7785;
  display: inline-block;
}

.emotion-4 {
  color: #6B7785;
  line-height: 40px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #E0E0E0;
  border-radius: 3px;
  cursor: pointer;
}

.emotion-9:not([disabled]):hover {
  background: #E0E0E0;
}

<div
  className="emotion-9"
>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="hour-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          Hrs
        </span>
        <span>
          <i
            className="pi pi-arrow-drop-down emotion-1"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
  <span
    className="emotion-4"
  >
    :
  </span>
  <div
    className="emotion-3"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "width": "100%",
        }
      }
    >
      <div
        className="emotion-2"
        data-test-id="minute-label"
        onClick={[Function]}
      >
        <span
          className="emotion-0"
        >
          mins
        </span>
        <i
          className="pi pi-arrow-drop-down emotion-1"
          style={
            Object {
              "transform": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
